### PR TITLE
NETBEANS-4852 add tooltip with code coverage stats

### PR DESCRIPTION
**Pull request by jherkel**
Add tooltip with code coverage stats to a code coverage overview window & code coverage bar
code cleanup

### CODE REVIEW

**Comment by lkishalmi** [comment]
Is it possible to create a JIRA improvement issue for this one, also a screenshot of the change would be welcome. That might go into the new features/improvements page later.

File: ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageBar.java [hunk 1]
```diff
@@ -55,9 +59,25 @@
...
+          if ((e.getChangeFlags() & HierarchyEvent.SHOWING_CHANGED) != 0) {
+            if (isShowing()) {
+              ToolTipManager.sharedInstance().registerComponent(CoverageBar.this);
+            } else {
+              ToolTipManager.sharedInstance().unregisterComponent(CoverageBar.this);
+            }
+          }
+        }
+      });      
+      updateUI();
```
Comment (junichi11) [inline ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageBar.java:80]: Please format. (4 spaces)

**Comment by lkishalmi** [comment]
@jherkel could you update this PR with the requested format change?

File: ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java [hunk 2]
```diff
@@ -168,6 +168,9 @@
         //    }
 
         totalCoverage.setCoveragePercentage(model.getTotalCoverage());
+        FileCoverageSummary summary = model.getCoverageSummary();
+        totalCoverage.setStats(summary.getLineCount(),summary.getExecutedLineCount(),
+          summary.getPartialCount(),summary.getInferredCount());
```
Comment (junichi11) [inline ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java:173]: Please format.

File: ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java [hunk 3]
```diff
@@ -361,6 +364,9 @@ void updateData(List<FileCoverageSummary> results) {
         model = new CoverageTableModel(results);
         table.setModel(model);
         totalCoverage.setCoveragePercentage(model.getTotalCoverage());
+        FileCoverageSummary summary = model.getCoverageSummary();
+        totalCoverage.setStats(summary.getLineCount(),summary.getExecutedLineCount(),
+          summary.getPartialCount(),summary.getInferredCount());
```
Comment (junichi11) [inline ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java:369]: Please format.

File: ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java [hunk 4]
```diff
@@ -399,19 +405,26 @@ public CoverageTableModel(List<FileCoverageSummary> results) {
             }
             this.results = results;
         }
+        
+        FileCoverageSummary getCoverageSummary() {
+          return total;
```
Comment (junichi11) [inline ide/gsf.codecoverage/src/org/netbeans/modules/gsf/codecoverage/CoverageReportTopComponent.java:410]: Please format.

**Comment by jherkel** [comment]
Sorry for the late response, I wasn't at my computer. Is there any page on wiki how to setup formatter? I will also create JIRA task for this pull request

**Comment by junichi11** [comment]
> Is there any page on wiki how to setup formatter?

Unfortunately, no but, maybe basically, it should be no problem with default settings.

File: apisupport/apisupport.ant/manifest.mf [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Manifest-Version: 1.0
 OpenIDE-Module: org.netbeans.modules.apisupport.ant
 OpenIDE-Module-Localizing-Bundle: org/netbeans/modules/apisupport/project/Bundle.properties
-OpenIDE-Module-Specification-Version: 2.82
+OpenIDE-Module-Specification-Version: 2.81
```
Comment (junichi11) [inline apisupport/apisupport.ant/manifest.mf:4]: Why?

Comment (jherkel) [reply apisupport/apisupport.ant/manifest.mf]: I accidentally made a wrong commit

**Comment by lkishalmi** [comment]
Well Travis seems to be happy with this.

**Pull request merged**